superx/information_extractors/item_info_extractor.py

```diff
@@ -124,23 +124,16 @@ def extract_information_from_parsed_xml(self, xml_info_child_node):
                 is_weighted = True
 
             unit_of_measure = self.standardize_weight_name(item.find('UnitQty').text)
-            # if item already in db then continue to next item
-            if bool(Product.query.filter_by(id=item_code).first()):
-                # adding current_branch_price to the db and commit
-                current_branch_price = BranchPrice(item_code=item_code, price=price, update_date=update_date)
-                db.session.add(current_branch_price)
-                db.session.commit()
-                continue
-
-            # adding current_product to the db and commit
-            current_product = Product(id=item_code, name=item_name, quantity=quantity, is_weighted=is_weighted,
-                                      unit_of_measure=unit_of_measure)
-            db.session.add(current_product)
-            db.session.commit()
-            # adding current_branch_price to the db and commit
-            current_branch_price = BranchPrice(item_code=item_code, price=price, update_date=update_date)
+            # if item is not in db then add it
+            if not bool(Product.query.filter_by(id=item_code).first()):
+                current_product = Product(id=item_code, name=item_name, quantity=quantity, is_weighted=is_weighted,
+                                          unit_of_measure=unit_of_measure)
+                db.session.add(current_product)
+
+            current_branch_price = BranchPrice(branch_id='', item_code=item_code, price=price, update_date=update_date)
```

Comment:
I still need to add a part that gets the branch id, for now i wrote it as empty

Reply:
We can merge it like this, but it won't work until you get the branch id.

Reply:
will update the pd today with branch id